### PR TITLE
remove feature toggle + add libraryPaneldefinition

### LINKED ISSUE #34485: remove trim/apply default feature toggle

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #34485

**Special notes for your reviewer**:

### CODE REVIEW

File: cue/data/gen.cue [hunk 1]
```diff
@@ -134,7 +134,11 @@ Family: scuemata.#Family & {
                     // The values depend on panel type
                     options: {...}
                     libraryPanel?: {
-                        name: string,
+                        description?: string | *""
+                        meta: {...}
```
Comment (hugohaggmark) [inline cue/data/gen.cue:138]: Not sure we need meta.

**Review by hugohaggmark — COMMENTED** [review]
LGTM! 🎉

**Review by marefr — COMMENTED** [review]
Added a comment. Have you or are you planning to document this new feature before the stable release? Besides feature documentation, dashboard HTTP API documentation probably needs an update due to these changes.

File: public/app/features/dashboard/components/ShareModal/ShareExport.tsx [hunk 2]
```diff
@@ -138,11 +137,9 @@ export class ShareExport extends PureComponent<Props, State> {
         <Field label="Export for sharing externally">
           <Switch value={shareExternally} onChange={this.onShareExternallyChange} />
         </Field>
-        {config.featureToggles.trimDefaults && (
-          <Field label="Export with trimed dashboard json">
-            <Switch value={trimDefaults} onChange={this.onTrimDefaultsChange} />
-          </Field>
-        )}
+        <Field label="Export with trimed dashboard json">
```
[the remark this answers is not in the document]
Comment (marefr) [reply public/app/features/dashboard/components/ShareModal/ShareExport.tsx:140]: @achatterjee-grafana do you have any input on this matter?

**Comment by ying-jeanne** [comment]
The panellibrary would have only uid when we export internally. when export externally, it would look like following. not sure whether we need to make the difference on schema definition or not.
      "libraryPanel": {
        "description": "",
        "meta": {
          "connectedDashboards": 3,
          "created": "2021-05-19T16:31:41+02:00",
          "createdBy": {
            "avatarUrl": "/avatar/46d229b033af06a191ff2267bca9ae56",
            "id": 1,
            "name": "admin"
          },
          "folderName": "General",
          "folderUid": "",
          "updated": "2021-05-19T16:31:41+02:00",
          "updatedBy": {
            "avatarUrl": "/avatar/46d229b033af06a191ff2267bca9ae56",
            "id": 1,
            "name": "admin"
          }
        }

File: public/app/features/dashboard/components/ShareModal/ShareExport.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (achatterjee-grafana) [reply public/app/features/dashboard/components/ShareModal/ShareExport.tsx:140]: I agree, "trim" seems too casual. Marcus's suggestion "Export with default values removed" is good.

**Comment by ying-jeanne** [comment]
> Added a comment. Have you or are you planning to document this new feature before the stable release? Besides feature documentation, dashboard HTTP API documentation probably needs an update due to these changes.

do you mean here https://docs.google.com/document/d/1WXhJI5jVliZc7eqv1Lrb9uxarLGz66mkOblqkAqYO-g/edit?ts=60a2b1bb# should we complete the md file for new features or tech writer would take care of it.

**Comment by ying-jeanne** [comment]
the md is updated, image for doc is updated separately here: https://github.com/grafana/website/pull/4537

File: docs/sources/dashboards/export-import.md [hunk 3]
```diff
@@ -25,6 +25,11 @@ add template variables for things like a metric prefix (use constant variable) a
 A template variable of the type `Constant` will automatically be hidden in
 the dashboard, and will also be added as a required input when the dashboard is imported.
 
+### Export with default values removed
```
Comment (achatterjee-grafana) [inline docs/sources/dashboards/export-import.md:28]:
```suggestion
### Export dashboard without default values
```

File: docs/sources/dashboards/export-import.md [hunk 4]
```diff
@@ -25,6 +25,11 @@ add template variables for things like a metric prefix (use constant variable) a
...
+
+If you want to export a dashboard without default values in order to reduce the exported json file size, it is a good idea to check the toggle "Export with default values removed".
```
Comment (achatterjee-grafana) [inline docs/sources/dashboards/export-import.md:30]:
```suggestion
To export a dashboard without the default values in order to reduce the exported JSON file size, check the toggle **Export with default values removed** option.
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added copy eedit suggestions.

File: docs/sources/dashboards/export-import.md [hunk 5]
```diff
@@ -25,6 +25,11 @@ add template variables for things like a metric prefix (use constant variable) a
...
+
+When import, the removed default values would be added automatically to the dashboard.
```
Comment (aknuds1) [inline docs/sources/dashboards/export-import.md:32]: Some language fixes, JSON should be written in all-caps as it's an acronym:
```suggestion
If you want to export a dashboard without default values in order to reduce the exported JSON file size, it is a good idea to check the toggle "Export with default values removed".

When importing, the removed default values will be added automatically to the dashboard.
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Saw some language issues in the docs.

File: docs/sources/dashboards/export-import.md (hunk 5 above)
Comment (achatterjee-grafana) [reply docs/sources/dashboards/export-import.md:32]: Agree, JSON should be all caps. Documentation best practice says you should avoid future tense if you can. How about: 

"During import, the removed default values automatically added back to the dashboard."

**Review by achatterjee-grafana — APPROVED** [review]

**Review by marefr — APPROVED** [review]
LGTM

File: docs/sources/dashboards/export-import.md [hunk 6]
```diff
@@ -25,6 +25,11 @@ add template variables for things like a metric prefix (use constant variable) a
 A template variable of the type `Constant` will automatically be hidden in
 the dashboard, and will also be added as a required input when the dashboard is imported.
 
+### Export dashboard without default values
+
+To export a dashboard without the default values in order to reduce the exported JSON file size, check the toggle **Export with default values removed** option.
+
+During import, the removed default values automatically added back to the dashboard.
```
Comment (oddlittlebird) [inline docs/sources/dashboards/export-import.md:32]:
```suggestion
During import, the removed default values automatically added back to the dashboard.

```

File: docs/sources/dashboards/export-import.md [hunk 7]
```diff
@@ -25,6 +25,11 @@ add template variables for things like a metric prefix (use constant variable) a
...
 ## Importing a dashboard
```
Comment (oddlittlebird) [inline docs/sources/dashboards/export-import.md:33]:
```suggestion
## Import a dashboard
```

**Review by hugohaggmark — APPROVED** [review]
LGTM 🎉 !

File: docs/sources/dashboards/export-import.md (hunk 5 above)
Comment (aknuds1) [reply docs/sources/dashboards/export-import.md]: There's a typo in @achatterjee-grafana's suggestion, it should probably be "During import, the removed default values are automatically added back to the dashboard.".

File: docs/sources/dashboards/export-import.md [hunk 8]
```diff
@@ -25,7 +25,13 @@ add template variables for things like a metric prefix (use constant variable) a
 A template variable of the type `Constant` will automatically be hidden in
 the dashboard, and will also be added as a required input when the dashboard is imported.
 
-## Importing a dashboard
+### Export dashboard without default values
+
+To export a dashboard without the default values in order to reduce the exported JSON file size, check the toggle **Export with default values removed** option.
+
+During import, the removed default values automatically added back to the dashboard.
```
Comment (aknuds1) [inline docs/sources/dashboards/export-import.md:32]:
```suggestion
During import, the removed default values are automatically added back to the dashboard.
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Looks good except there's still a documentation typo.

**Review by aknuds1 — APPROVED** [review]
LGTM

**Comment by ying-jeanne** [comment]
hi @marefr @achatterjee-grafana could you take a second look at the api/dashboard.md that i added for the new endpoint (trim input dashboard json) thanks.

File: docs/sources/http_api/dashboard.md [hunk 9]
```diff
@@ -541,3 +541,97 @@ Content-Length: 97
...
+## Trim input dashboard
+
+`POST /api/dashboards/trim`
+
+Will remove the default values from input dashboard JSON.
+
+**Example Request for trimming dashboard JSON**:
+
+```http
+POST /api/dashboards/uid/trim HTTP/1.1
```
Comment (marefr) [inline docs/sources/http_api/dashboard.md:554]: Uid shouldn't be included here right?

File: docs/sources/http_api/dashboard.md [hunk 10]
```diff
@@ -541,3 +541,97 @@ Content-Length: 97
...
+Accept: application/json
+Content-Type: application/json
+Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
+
+{
+    "meta": {
```
Comment (marefr) [inline docs/sources/http_api/dashboard.md:560]: Is meta really included/required in the request?

File: docs/sources/http_api/dashboard.md [hunk 11]
```diff
@@ -541,3 +541,97 @@ Content-Length: 97
   "status": "multiple-slugs-exists"
 }
 ```
+
+## Trim input dashboard
```
Comment (marefr) [inline docs/sources/http_api/dashboard.md:545]: Would move this section above the deprecated section. 

Again, should we refer to trim or name this Remove default values in dashboard?

**Review by marefr — COMMENTED** [review]
Added some comments/questions

File: docs/sources/http_api/dashboard.md (hunk 9 above)
Comment (ying-jeanne) [reply docs/sources/http_api/dashboard.md:554]: no, we don't read from database, just based on the input, uid would not be needed

File: docs/sources/http_api/dashboard.md (hunk 11 above)
Comment (ying-jeanne) [reply docs/sources/http_api/dashboard.md:545]: you are right, remove default values in dashboard would sounds better.

File: docs/sources/http_api/dashboard.md (hunk 10 above)
Comment (ying-jeanne) [reply docs/sources/http_api/dashboard.md:560]: it is not required, but allowed. would you recommend not put it in the example?

**Pull request merged**